### PR TITLE
[Bugfix] Preserve channel-last video tensor layout

### LINKED ISSUE #5417: [Bug]: Video encoding corrupts channel-last tensors with 3 or 4 frames
### Your current environment

Current `main` at `2bb618903afc9dc29a1f29578dcac31caa047ad5`.

### Your code version

vLLM-Omni `main`.

### Describe the bug

The OpenAI-compatible video encoding path misclassifies channel-last Torch tensors when the generated video contains exactly 3 or 4 frames.

`_normalize_video_tensor()` treats every rank-4 tensor whose first dimension is 3 or 4 as `[C, F, H, W]`. However, generated video tensors are also accepted in the standard channel-last layout `[F, H, W, C]`. For a 3- or 4-frame RGB video, both the frame count and channel count are 3 or 4, so the current first-dimension-only check transposes an already channel-last tensor.

For example, a tensor with shape `[4, 2, 5, 3]` is transformed into `[2, 5, 3, 4]`. The downstream encoder then interprets it as two RGBA frames and drops the fourth channel, producing the wrong frame count and spatial shape.

This affects both synchronous MP4 encoding and fragmented MP4 streaming because both paths call `_coerce_video_to_uint8_frames()`.

Minimal reproduction:

```python
import torch
from vllm_omni.entrypoints.openai.video_api_utils import _coerce_video_to_uint8_frames

video = torch.zeros((4, 2, 5, 3), dtype=torch.uint8)
frames = _coerce_video_to_uint8_frames(video)
assert frames.shape == (4, 2, 5, 3)
```

Current result: `frames.shape == (2, 5, 3, 3)`.

Expected result: channel-last tensors preserve their frame count and shape. Channel-first `[C, F, H, W]` tensors should remain supported.

### Proposed fix

Use the shared NumPy layout normalization for Torch tensors as well. That helper already gives an explicit channel-last dimension precedence in ambiguous shapes and only converts channel-first layouts when the last dimension is not a valid channel count.

### PR DESCRIPTION
**Pull request by whisper-la**
## Purpose

Fix video encoding for channel-last PyTorch tensors containing exactly 3 or 4 frames.

Previously, `_normalize_video_tensor` treated any 4D tensor whose first dimension was 3 or 4 as channel-first. Therefore, valid `[F, H, W, C]` tensors with 3 or 4 frames were incorrectly permuted and encoded with the wrong frame or spatial dimensions.

This change delegates tensor layout detection to the existing shared NumPy normalization logic. It preserves channel-last tensors while retaining support for channel-first `[C, F, H, W]` tensors.

Fixes #5417.

## Test Plan

Run the targeted regression tests:

```bash
python -m pytest -s -v \
  tests/entrypoints/openai_api/test_video_api_utils.py::test_coerce_video_preserves_channel_last_tensor_with_few_frames \
  tests/entrypoints/openai_api/test_video_api_utils.py::test_coerce_video_converts_channel_first_tensor
```
  Run the complete related test file:
```bash
python -m pytest -s -v \
  tests/entrypoints/openai_api/test_video_api_utils.py
```

Hardware: NVIDIA H100
vLLM Version: 0.26.0
vLLM-Omni Commit: 1b9f7ea882325e50ccb0aad1fecbb15a5854328f

## Test Result
Targeted video layout regression tests: 3 passed
Complete video API utility test suite: 10 passed
No test failures
The targeted tests verify that:
Channel-last RGB tensors with 3 frames preserve their original layout and values.
Channel-last RGB tensors with 4 frames preserve their original layout and values.
Channel-first tensors are still converted correctly to channel-last format.

### CODE REVIEW

**Comment by lishunyang12** [comment]
Thanks for your contribution. Already fixed by the issue author.
